dsl/src/main/groovy/org/cristalise/dsl/persistency/outcome/SchemaDelegate.groovy

```diff
@@ -97,16 +96,37 @@ class SchemaDelegate {
             }
         }
     }
-    
+
     private boolean hasRangeConstraints(Attribute a) {
         return a.minInclusive != null || a.maxInclusive != null || a.minExclusive != null || a.maxExclusive != null
     }
 
+    private boolean hasRestrictions(Attribute a) {
+        return a.values || a.pattern || hasRangeConstraints(a) || a.precision != null || a.scale != null
+    }
+
+    /**
+     * Checks whether the field has a restriction/attributes/unit or not, because the type of the element 
+     * is either specified in the 'type' attribute or in the restriction as 'base'
+     * 
+     * @param f the actual field to check
+     * @return the type if the field has nor restriction otherwise an empty string
```

Comment:
Replace 
`the type if the field has nor restriction otherwise an empty string` with 
`the type of the field has no restriction otherwise an empty string`
`_________^__________________^`